### PR TITLE
PartnerTenantType on Organization

### PR DESCRIPTION
**Pull request by akshukla25**

### CODE REVIEW

**Comment by Lauragra** [comment]
@akshukla25 , please follow the Get started steps to set yourself up to be a contributor. This PR needs to go through a content review process; it currently has a validation error.

https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211776/Get-started

**Comment by akshukla25** [comment]
> @akshukla25 , please follow the Get started steps to set yourself up to be a contributor. This PR needs to go through a content review process; it currently has a validation error.
> 
> https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211776/Get-started

completed these steps.

**Comment by Lauragra** [comment]
> > @akshukla25 , please follow the Get started steps to set yourself up to be a contributor. This PR needs to go through a content review process; it currently has a validation error.
> > https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211776/Get-started
> 
> completed these steps.

Thank you. When you're ready for us to review your PR, please set the **ready for content review** label. It looks like this PR is failing validation because we do not have a resource topic for the partnerTenantType resource - please investigate.

**Comment by akshukla25** [comment]
> > > @akshukla25 , please follow the Get started steps to set yourself up to be a contributor. This PR needs to go through a content review process; it currently has a validation error.
> > > https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211776/Get-started
> > 
> > 
> > completed these steps.
> 
> Thank you. When you're ready for us to review your PR, please set the **ready for content review** label. It looks like this PR is failing validation because we do not have a resource topic for the partnerTenantType resource - please investigate.

Could you please help me understand what is resource topic here and what am I missing?

**Comment by Lauragra** [comment]
> > > > @akshukla25 , please follow the Get started steps to set yourself up to be a contributor. This PR needs to go through a content review process; it currently has a validation error.
> > > > https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211776/Get-started
> > > 
> > > 
> > > completed these steps.
> > 
> > 
> > Thank you. When you're ready for us to review your PR, please set the **ready for content review** label. It looks like this PR is failing validation because we do not have a resource topic for the partnerTenantType resource - please investigate.
> 
> Could you please help me understand what is resource topic here and what am I missing?

If there is a schema PR associated with this content change, the associated stub topics should have been generated as part of that PR. You can find information about resource topics in the CDK, here: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/344038/Reference-topic-templates?anchor=template-for-an-entity

**Comment by FaithOmbongi** [comment]
Hi @akshukla25 - I've come across this PR and see it's related to an Identity workload. Because the schema is already live, kindly ping me internally so I can guide you to close this docs gap.

File: api-reference/beta/resources/organization.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ This resource lets you add your own data to custom properties using [extensions]
 | marketingNotificationEmails | String collection | Not nullable. |
 | onPremisesLastSyncDateTime | DateTimeOffset | The time and date at which the tenant was last synced with the on-premises directory. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`.|
 | onPremisesSyncEnabled | Boolean | `true` if this object is synced from an on-premises directory; `false` if this object was originally synced from an on-premises directory but is no longer synced; Nullable. `null` if this object has never been synced from an on-premises directory (default). |
+|partnerTenantType|partnerTenantType| Type of the partner tenant.The possible values are: `microsoftSupport`, `syndicatePartner`, `breadthPartner`, `breadthPartnerDelegatedAdmin`, `resellerPartnerDelegatedAdmin`, `valueAddedResellerPartnerDelegatedAdmin`, `unknownFutureValue`.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/organization.md:65]: - It's not clear what "partner tenant" is. Are these Microsoft Partners? Can we make it clearer?
- Follow Option 2 of the enumeration guidance to document the enums in a H3 section after the properties table. The anchor link I added will point to it.
  - This fix will also resolve the validation error.

```suggestion
|partnerTenantType|partnerTenantType| Type of the partner tenant. The possible values are: `microsoftSupport`, `syndicatePartner`, `breadthPartner`, `breadthPartnerDelegatedAdmin`, `resellerPartnerDelegatedAdmin`, `valueAddedResellerPartnerDelegatedAdmin`, `unknownFutureValue`. Nullable. For more information about the possible types, see [partnerTenantType values](#partnertenanttype-values).|
```

File: api-reference/v1.0/api/organization-get.md [hunk 2]
```diff
@@ -125,6 +125,7 @@ Content-type: application/json
     "marketingNotificationEmails": [],
     "onPremisesLastSyncDateTime": null,
     "onPremisesSyncEnabled": null,
+    "partnerTenantType": "String",
```
Comment (FaithOmbongi) [inline api-reference/v1.0/api/organization-get.md:128]: GLOBAL: Change to a real value or `null`

File: api-reference/v1.0/resources/organization.md [hunk 3]
```diff
@@ -55,6 +55,7 @@ This resource lets you add your own data to custom properties using [extensions]
 | marketingNotificationEmails | String collection | Not nullable. |
 | onPremisesLastSyncDateTime | DateTimeOffset | The time and date at which the tenant was last synced with the on-premises directory. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`. Read-only.|
 | onPremisesSyncEnabled | Boolean | `true` if this object is synced from an on-premises directory; `false` if this object was originally synced from an on-premises directory but is no longer synced. Nullable. `null` if this object has never been synced from an on-premises directory (default). |
+| partnerTenantType | Type of the partner tenant. The possible values are: `microsoftSupport`, `syndicatePartner`, `breadthPartner`, `breadthPartnerDelegatedAdmin`, `resellerPartnerDelegatedAdmin`, `valueAddedResellerPartnerDelegatedAdmin`, `unknownFutureValue`.|
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/organization.md:58]: Apply the same suggestions provided in the beta doc

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @akshukla25 for your work on this PR.
The two major pieces of feedback:

- The definition of partnerTenantType isn't clear. Please check and address.
- Document the partnerTenantType enumeration as a H3 after the properties table. That should resolve your validation errors. I've shared with you the guidance for documenting enums.
- GLOBAL: For all API docs where you've added the property to the payload, instead of "String", replace with one of the possible values or `null`.

File: api-reference/beta/resources/organization.md [hunk 4]
```diff
@@ -62,6 +62,7 @@ This resource lets you add your own data to custom properties using [extensions]
...
 | onPremisesLastSyncDateTime | DateTimeOffset | The time and date at which the tenant was last synced with the on-premises directory. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`.|
 | onPremisesSyncEnabled | Boolean | `true` if this object is synced from an on-premises directory; `false` if this object was originally synced from an on-premises directory but is no longer synced; Nullable. `null` if this object has never been synced from an on-premises directory (default). |
+| partnerTenantType | partnerTenantType | Type of the partner tenant that has a partnership with the given tenant where user's group in a partner tenant are given an ability to manage the given tenant. The possible values are: `microsoftSupport`, `syndicatePartner`, `breadthPartner`, `breadthPartnerDelegatedAdmin`, `resellerPartnerDelegatedAdmin`, `valueAddedResellerPartnerDelegatedAdmin`, `unknownFutureValue`. Nullable. For more information about the possible types, see [partnerTenantType values](#partnertenanttype-values).|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/organization.md:65]: Is partnerTenantType returned for the Microsoft Partner or is it returned for the customer that has a relationship with the Microsoft Partner?
```suggestion
| partnerTenantType | partnerTenantType | Type of the tenant that has a delegated partnership with the given tenant where user's group in a partner tenant are given the ability to manage the given tenant. The possible values are: `microsoftSupport`, `syndicatePartner`, `breadthPartner`, `breadthPartnerDelegatedAdmin`, `resellerPartnerDelegatedAdmin`, `valueAddedResellerPartnerDelegatedAdmin`, `unknownFutureValue`. Nullable. For more information about the possible types, see [partnerTenantType values](#partnertenanttype-values).|
```

File: api-reference/beta/resources/organization.md [hunk 5]
```diff
@@ -73,6 +74,20 @@ This resource lets you add your own data to custom properties using [extensions]
 | technicalNotificationMails |String collection | Not nullable. |
 | verifiedDomains | [verifiedDomain](verifieddomain.md) collection|The collection of domains associated with this tenant. Not nullable. |
 
+### partnerTenantType values
+
+The following table lists the members of an [evolvable enumeration](/graph/best-practices-concept#handling-future-members-in-evolvable-enumerations). You must use the `Prefer: include-unknown-enum-members` request header to get the following values in this evolvable enum: `ResellerPartnerDelegatedAdmin`, `ValueAddedResellerPartnerDelegatedAdmin`.
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/organization.md:79]: As per your schema and the description above, there's no value after the `unknownFutureValue`; So Example 2 of Option 2 in the guidance was the suitable one.
```suggestion
```

File: api-reference/beta/resources/organization.md [hunk 6]
```diff
@@ -73,6 +74,20 @@ This resource lets you add your own data to custom properties using [extensions]
...
+| BreadthPartner | Partner has the ability to provide administrative support for their customers. However, the partner is not allowed to resell to or request delegated admin permissions from customers. |
+| BreadthPartnerDelegatedAdmin | Similar to Breadth Parther except they can request for delegated admin permissions (DAP) from customer tenants. |
+| SyndicatePartner | Partner that exclusively resells and manages O365 and Intune for their customers. The customer cannot buy additional direct subscriptions from Microsoft or from other partners. |
+| ResellerPartnerDelegatedAdmin | Partner that can resell and manage O365 and Intune for their customers. There is no exclusivity clause unlike in the case of SyndicationPartners. The customer can buy additional direct subscriptions from Microsoft or from other partners. |
+| ValueAddedResellerPartnerDelegatedAdmin | ValueAddedReseller are tier 2 partners, they will themselves be a partner while having a relationship with another partner (generally a reseller), that's why they are "value added reseller". |
+| unknownFutureValue | Evolvable enumeration sentinel value. Do not use. |
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/organization.md:89]: Exact syntax that's used in the API and schema please.
```suggestion
| microsoftSupport | Legacy support tenant of Microsoft which was used to provide administrative support to all the customer tenants. |
| breadthPartner | Partner has the ability to provide administrative support for their customers. However, the partner is not allowed to resell to or request delegated admin permissions from customers. |
| breadthPartnerDelegatedAdmin | Similar to Breadth Parther except they can request for delegated admin permissions (DAP) from customer tenants. |
| syndicatePartner | Partner that exclusively resells and manages O365 and Intune for their customers. The customer cannot buy additional direct subscriptions from Microsoft or from other partners. |
| resellerPartnerDelegatedAdmin | Partner that can resell and manage O365 and Intune for their customers. There is no exclusivity clause unlike in the case of SyndicationPartners. The customer can buy additional direct subscriptions from Microsoft or from other partners. |
| valueAddedResellerPartnerDelegatedAdmin | ValueAddedReseller are tier 2 partners, they will themselves be a partner while having a relationship with another partner (generally a reseller), that's why they are "value added reseller". |
| unknownFutureValue | Evolvable enumeration sentinel value. Do not use. |
```

File: api-reference/v1.0/resources/organization.md [hunk 7]
```diff
@@ -66,6 +67,20 @@ This resource lets you add your own data to custom properties using [extensions]
 | technicalNotificationMails | String collection | Not nullable. |
 | verifiedDomains | [VerifiedDomain](verifieddomain.md) collection | The collection of domains associated with this tenant. Not nullable. |
 
+### partnerTenantType values
+
+The following table lists the members of an [evolvable enumeration](/graph/best-practices-concept#handling-future-members-in-evolvable-enumerations). You must use the `Prefer: include-unknown-enum-members` request header to get the following values in this evolvable enum: `ResellerPartnerDelegatedAdmin`, `ValueAddedResellerPartnerDelegatedAdmin`.
+
+| Member | Description |
```
Comment (FaithOmbongi) [inline api-reference/v1.0/resources/organization.md:74]: Apply the same suggestions I shared in beta

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]

**Review by FaithOmbongi — APPROVED** [review]
LGTM. Approved!

**Pull request merged**